### ISSUE TITLE
Bug: Assign token with empty serial resulting in actual assign a random token

### ISSUE THREAD

**Issue by quynh-axiadids** [privacyidea/privacyidea#1292]
# What did you try to do?

Click on Assign Token in User Details tab without entering any serial/pin in the input fields

# What outcome did you expect?

Fails to assign token due to serial is empty

# What outcome did you experience?

A random token is actually assigned. Sometimes fails to assign due to 'the' token already assigned to a user/self.

# Configuration

* privacyIDEA Version: 2.21.4

* Installation method: from source

## more details: 

* OS: Amazon Linux

* Webserver: Nginx

* Tokendatabase: postgres

# Log file

**Turn on debug level and take a look at the privacyidea.log!**
**If appropriate, attach the log file.**

This is the debug I got from adding some some verbose in privacyidea/lib/token.py
```
@log_with(log)
def assign_token(serial, user, pin=None, encrypt_pin=False):
    tokenobject_list = get_tokens(serial=serial)

    if not tokenobject_list:
        log.warning("no tokens found with serial: {0!r}".format(serial))
        raise TokenAdminError("no token found!", id=1102)

    print(">>> DEBUG: serial param " + serial)
    for t in tokenobject_list:
        print(">>> DEBUG: " + t.token.serial)
```
Output
```
>>> DEBUG: serial param
>>> DEBUG: OATH00094407
>>> DEBUG: OATH0001E2E5
>>> DEBUG: OATH00029524
>>> DEBUG: OATH00031834
>>> DEBUG: PISP00007061
```

The param serial is empty, but the tokenobject_list seems to contains all tokens in the database.
Seems like there is a bug in assigning a empty serial from the UI, or is this expected by design?

Thank you.

**Comment by cornelinux** [comment]
@quynh-axiadids  This is fixed in 2.23. 
We changed this in ``get_tokens`` this way, that serials will by default only do exact matches:
See https://github.com/privacyidea/privacyidea/blob/master/privacyidea/lib/token.py#L214

Before this a wildcard match was performed, which could lead to such results but also to performance issues. We recommend to update to 2.23.2 or 2.23.3.

**Comment by quynh-axiadids** [comment]
Thanks for your feedback!
